app/org/sagebionetworks/bridge/play/controllers/ApplicationController.java

```diff
@@ -18,15 +27,34 @@ public Result loadApp() throws Exception {
         return ok(views.html.index.render());
     }
 
-    public Result verifyEmail(String study) {
-        Study studyObj = studyService.getStudy(study);
+    public Result verifyEmail(String studyId) {
+        if (isBlank(studyId)) {
+            throw new BadRequestException("study parameter is required");
+        }
+        Study study = studyService.getStudy(studyId);
         return ok(views.html.verifyEmail.render(ASSETS_HOST, ASSETS_BUILD,
-                StringEscapeUtils.escapeHtml4(studyObj.getName()), studyObj.getSupportEmail()));
+                StringEscapeUtils.escapeHtml4(study.getName()), study.getSupportEmail()));
     }
 
-    public Result resetPassword(String study) {
-        Study studyObj = studyService.getStudy(study);
+    public Result resetPassword(String studyId) {
+        if (isBlank(studyId)) {
+            throw new BadRequestException("studyId parameter is required");
+        }
+        Study study = studyService.getStudy(studyId);
         return ok(views.html.resetPassword.render(ASSETS_HOST, ASSETS_BUILD,
-                StringEscapeUtils.escapeHtml4(studyObj.getName()), studyObj.getSupportEmail()));
+                StringEscapeUtils.escapeHtml4(study.getName()), study.getSupportEmail()));
+    }
+    
+    public Result startSession(String studyId, String email, String token) {
+        if (isBlank(studyId)) {
+            throw new BadRequestException("study ID is required");
+        }
+        SignIn signIn = new SignIn(studyId, email, null, token);
```

Comment:
This is receiving the link directly from the email where the values are parameters, not a JSON post. It cannot be combined with the other method. It is here so that if an email is sent and the context is such that the app does not intercept the link, you will get back a session. It might be useful for testing, it's not useful in production because there's not the user could do with the session outside of the app.